tests/e2e/taqueria-plugin-ipfs-pinata.spec.ts

```diff
@@ -1,132 +1,186 @@
-		await exec(`echo "pinataJwtToken=${JWT}" > ${taqueriaProjectPath}/.env`);
-	}
-	if (process.env.UNLIMITED_PINATA_TOKEN) {
-		JWT = process.env.UNLIMITED_PINATA_TOKEN;
-		await exec(`echo "pinataJwtToken=${JWT}" > ${taqueriaProjectPath}/.env`);
-		await exec(`cat ${taqueriaProjectPath}/.env`);
-	} else {
-		// The .env file should be in the root directory of the taqueria project
-		// this just makes sure it gets into the test directory for use
-		await exec(`cp ../.env ${taqueriaProjectPath}/.env`);
-	}
-}
-
-describe('e2e testing for the IPFS Pinata plugin with no JWT', () => {
-	beforeAll(async () => {
-		await generateTestProject(taqueriaProjectPath, ['ipfs-pinata']);
-		// TODO: This can removed after this is resolved:
-		// https://github.com/ecadlabs/taqueria/issues/528
-		try {
-			await exec(`taq -p ${taqueriaProjectPath}`);
-		} catch (_) {}
-	});
+import { prepareEnvironment } from '@gmrchk/cli-testing-library';
 
-	// Skipping due to help output changing
-	test.skip('ipfs pinata plugin should show the correct help contents', async () => {
-		const publishRun = await exec(`taq --help`, { cwd: taqueriaProjectPath });
-		expect(publishRun.stdout).toBe(contents.helpContentsIPFSPinataPlugin);
-	});
+describe('IPFS Pinata plugin E2E tests for Taqueria CLI', () => {
+	test('publish will offer contextual help', async () => {
+		const { execute, cleanup, exists } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
+		await exists('./test-project/.taq/config.json');
+		const {} = await execute('taq', 'install ../taqueria-plugin-ipfs-pinata', './test-project');
+		await exists('./test-project/node_modules/@taqueria/plugin-ipfs-pinata/index.js');
 
-	// Skipping due to help output changing
-	test.skip('ipfs pinata plugin should show the correct help contents for publish', async () => {
-		const publishRun = await exec(`taq publish --help`, { cwd: taqueriaProjectPath });
-		expect(publishRun.stdout).toBe(contents.helpContentsIPFSPinataPluginPublish);
-	});
+		const { stdout: stdout2 } = await execute('taq', 'publish --help', './test-project');
+		expect(stdout2).toEqual(expect.arrayContaining(['Upload and pin files using your pinata account.']));
 
-	// Skipping due to help output changing
-	test.skip('ipfs pinata plugin should show the correct help contents for pin', async () => {
-		const publishRun = await exec(`taq pin --help`, { cwd: taqueriaProjectPath });
-		expect(publishRun.stdout).toBe(contents.helpContentsIPFSPinataPluginPin);
+		await cleanup();
 	});
 
-	test('ipfs pinata plugin should warn for publish if no env jwt token exists', async () => {
-		const publishRun = await exec(`taq publish ./ipfs/0001.txt`, { cwd: taqueriaProjectPath });
-		expect(publishRun.stderr).toMatch(/jwt *token.*not *found/i);
+	test('pin will offer contextual help', async () => {
+		const { execute, cleanup, exists } = await prepareEnvironment();
+		const {} = await execute('taq', 'init test-project');
```

Comment:
You can elide the assignment here and below.

Reply:
elided